### PR TITLE
Fix Node Startup Issue

### PR DESCRIPTION
**Pull request by benthecarman**
Caused by #1651, that fixed the comparison issue, but we needed keep dropping the leading zeros for to properly map to a `BigInt`.

Did a couple other small fixes like a typo, and put `getBestBlockHeader` all inside a `Future` so any errors thrown should propagate up

### CODE REVIEW

File: chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala [hunk 1]
```diff
@@ -54,26 +54,28 @@ case class ChainHandler(
...
-        val errMsg = s"Did not find blockchain with work $maxWork"
-        logger.error(errMsg)
-        throw new RuntimeException(errMsg)
-      case chain +: Vector() =>
-        chain.tip
-      case chain +: rest =>
-        logger.warn(
-          s"We have multiple competing blockchains: ${(chain +: rest).map(_.tip.hashBE.hex).mkString(", ")}")
-        chain.tip
+    Future {
```
Comment (Christewart) [inline chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala:57]: Do we want to add an invariant on `ChainHandler` to say 

```scala
require(blockchains.nonEmpty)
```

**Review by Christewart — APPROVED** [review]

File: chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala (hunk 1 above)
Comment (benthecarman) [reply chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala:57]: We can't actually do that, it causes most of our chain tests to fail.

**Comment by Christewart** [comment]
This didn't seem to work for me.

The server does not bind:

> ./bitcoin-s-cli getblockcount
Connection refused (Connection refused)

I also don't have any logs related to `chain-verification` after this one:

> 2020-07-20T16:39:12 DEBUG [chain-verification] Querying for best block hash

**Comment by Christewart** [comment]
This still seems not to work for me on 592e1af21c7a7cae0bdab1bf5d86e99f561074e5

Here are the logs when trying to sync my testnet node. If you look closely you see the last thing that is logged that is relevant to this PR is 

>2020-07-21T22:22:19 DEBUG [chain-verification] Querying for best block hash 

So it appears be dying in the chain logic again?

>0-07-21T22:22:19 DEBUG [P2P] Unaligned bytes after this: 0
2020-07-21T22:22:19 DEBUG [P2P] Received message=version from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Initializing
2020-07-21T22:22:19 DEBUG [P2P] Sending msg=verack to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-21T22:22:19 DEBUG [P2P] Received message=verack from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Initializing
2020-07-21T22:22:19 INFO [P2P] Our peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) has been initialized
2020-07-21T22:22:19 INFO [P2P] Our node has been full started. It took=4511ms
2020-07-21T22:22:19 DEBUG [chain-verification] Querying for best block hash
2020-07-21T22:22:19 DEBUG [P2P] Received message=sendheaders from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal
2020-07-21T22:22:19 DEBUG [P2P] Sending msg=sendheaders to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-21T22:22:19 DEBUG [P2P] Rec

**Pull request merged**